src/api/createDb.js

```diff
@@ -115,27 +116,34 @@ const createDb = levelDb => {
     await levelDb.put('nodeState', JSON.stringify(state));
   };
 
-  const storePeriods = async submissions => {
-    const dbOpsBatch = levelDb.batch();
-    await Promise.all(
-      submissions.map(submission => {
-        const key = `period!${submission.periodStart}`;
-        return getNullable(key).then(existingRecords => {
-          const records = existingRecords || [];
-          delete submission.periodStart;
-          records.push(submission);
-          dbOpsBatch.put(key, JSON.stringify(records));
-        });
-      })
+  const getPeriodData = periodStart => getNullable(`period!${periodStart}`);
+
+  const getPeriodDataByBlocksRoot = blocksRoot => {
+    return getNullable(`period!${blocksRoot}`).then(periodDataKey => {
+      if (!periodDataKey) return null;
+      return getNullable(periodDataKey);
+    });
+  };
+
+  const storeSubmission = async (periodStartHeight, submission) => {
+    const existingRecord = await getPeriodDataByBlocksRoot(
+      submission.blocksRoot
     );
+    // skip saving if record with the same root exists, otherwise overwrite
+    if (existingRecord && existingRecord.blocksRoot === submission.blocksRoot) {
+      return Promise.resolve();
+    }
+
+    const dbOpsBatch = levelDb.batch();
+    const key = `period!${periodStartHeight}`;
+    dbOpsBatch.put(`period!${submission.blocksRoot}`, key);
+    dbOpsBatch.put(key, JSON.stringify(submission));
```

Comment:
storing a single submission per period now. New submission for the same starting height will overwrite, unless having the same blocksRoot